frontend/src/components/authchooser/index.tsx

```diff
@@ -90,24 +97,37 @@ function AuthChooser(){
         .catch((err) => {
           if (!cancelled) {
             console.debug('Requiring token as testing auth failed:', err);
-            useToken = true;
+
+            // Ideally we'd only not assign the error if it was 401 or 403 (so we let the logic
+            // proceed to request a token), but let's first check whether this is all we get
+            // from clusters that require a token.
+            if ([408, 504, 502].includes(err.status)) {
+              errorObj = err;
+            }
+
             setTestingAuth(false);
           }
         })
         .finally(() => {
           if (!cancelled) {
+            cancelled = true;
+            setTestingAuth(false);
+
+            if (!!errorObj) {
+              setError(errorObj);
+              return;
+            }
+
             cluster.useToken = useToken;
             dispatch(setConfig({clusters: {...clusters}}));
             // If we don't require a token, then we just move to the attempted URL or root.
             if (!useToken) {
               history.replace(from);
             }
 
-            setTestingAuth(false);
-
             // If we reach this point, then we know whether or not we need a token. If we don't,
             // just redirect.
-            if (!cluster.useToken) {
+            if (cluster.useToken === false) {
```

Comment:
How about just?

```suggestion
            if (useToken === false) {
```

Reply:
This is deliberate. I want to make sure we're analysing the `cluster`'s property and not the variable. This is usually safer code in this context, as if later we don't assign the `useToken` directly, we still rely on what the cluster's property is.